Automattic/newspack-plugin
# feat: add option to choose which RAS meta fields to sync

### All Submissions:






### Changes proposed in this Pull Request:



Adds a new option under **Reader Activation > Advanced Settings** that allows admins to choose whether contact data is synced to the connected ESP, and which metadata fields should be synced. Also tweaks some UI component styles and some specific styles within the Reader Activation settings page.

Supersedes https://github.com/Automattic/newspack-network/pull/61.

### How to test the changes in this Pull Request:

1. Visit **Reader Activation > Advanced Settings** and confirm there's a new section grouping all settings related to ESP syncing behavior:

<img width="1047" alt="Screenshot 2024-03-15 at 12 09 34 PM" src="https://github.com/Automattic/newspack-plugin/assets/2230142/fc485b74-8d6a-4291-872f-82f33cd804a4">

2. Confirm that the the "Sync contacts to ESP" toggle is ON by default, and all metadata fields are enabled by default.
3. Smoke test contact syncing via a subscription purchase without changing any settings and confirm behavior is unchanged (all fields in [Newspack_Newsletters class](https://github.com/Automattic/newspack-plugin/blob/trunk/includes/plugins/class-newspack-newsletters.php#L26-L51) are synced).
4. De-select some fields and save. Confirm that the settings are persisted after refreshing the page.
5. Re-test a subscription purchase in a new session and confirm that only the fields kept selected in step 4 are synced.
6. Toggle OFF the "Sync contacts to ESP" option and re-test another subscription purchase. This time confirm that no contact data is synced to the ESP.

### Other information:




---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206151442802191
